components/shapediver/appbuilder/widgets/AppBuilderControlsWidgetComponent.tsx

```diff
@@ -0,0 +1,159 @@
+import {AppBuilderActionFromType} from "@AppBuilderShared/components/shapediver/appbuilder/actions/AppBuilderActionFromType";
+import {ComponentContext} from "@AppBuilderShared/context/ComponentContext";
+import {useExports} from "@AppBuilderShared/hooks/shapediver/parameters/useExports";
+import {useParameters} from "@AppBuilderShared/hooks/shapediver/parameters/useParameters";
+import {
+	getExportComponent,
+	getParameterComponent,
+} from "@AppBuilderShared/types/components/shapediver/componentTypes";
+import {PropsParameter} from "@AppBuilderShared/types/components/shapediver/propsParameter";
+import {
+	IAppBuilderControlActionRef,
+	IAppBuilderControlExportRef,
+	IAppBuilderControlParameterRef,
+	IAppBuilderWidgetPropsControls,
+} from "@AppBuilderShared/types/shapediver/appbuilder";
+import {Paper, Stack} from "@mantine/core";
+import React, {ReactElement, useContext, useMemo} from "react";
+type Props = IAppBuilderWidgetPropsControls & {
+	namespace: string;
+};
+
+export default function AppBuilderControlsWidgetComponent(props: Props) {
+	const {controls = [], namespace} = props;
+	const componentContext = useContext(ComponentContext);
+
+	// Convert parameter and export references to component props
+	const parameterProps: PropsParameter[] = useMemo(
+		() =>
+			controls
+				.filter((control) => control.type === "parameter")
+				.map((control) => {
+					const p = control.props as IAppBuilderControlParameterRef;
+					return {
+						namespace: p.sessionId ?? namespace,
+						parameterId: p.name,
+						disableIfDirty: p.disableIfDirty,
+						acceptRejectMode: p.acceptRejectMode,
+						overrides: p.overrides,
+					};
+				}),
+		[controls, namespace],
+	);
+
+	const exportProps = useMemo(
+		() => ({
+			namespace,
+			exports: controls
+				.filter((control) => control.type === "export")
+				.map((control) => {
+					const p = control.props as IAppBuilderControlExportRef;
+					return {
+						exportId: p.name,
+						overrides: p.overrides,
+					};
+				}),
+		}),
+		[controls, namespace],
+	);
```

Comment:
@LemToUp please refactor this at the same time as this comment: https://github.com/shapediver/AppBuilderShared/pull/156/files/3e3943f38372b313ab2a7b609aa00cea56f45c50..0e67db6a49382ab56fa5e9f71c295d58cce008fe#r2300280529
Most importantly, the `namespace` property needs to move inside, like we do for `parameterProps`. Each export might use a different `namespace` (aka `sessionId`)